dough-inventories/src/main/java/io/github/bakedlibs/dough/inventory/InvUtils.java

```diff
@@ -223,4 +224,88 @@ public static boolean removeItem(@Nonnull Inventory inv, int amount, boolean rep
         return false;
     }
 
+    /**
+     * Fills the borders of an {@link Inventory}
+     *
+     * @param inv The target {@link Inventory}
+     * @param item Item to use for filling
+     */
+    public static void fillBorder(@Nonnull Inventory inv, @Nonnull ItemStack item) {
+        Validate.notNull(inv, "Inventory must not be null");
+        Validate.notNull(item, "Item must not be null");
+
+        int size = inv.getSize();
+        int rows = (size + 1) / 9;
+
+        // Fill top
+        for (int i = 0; i < 9; i++) {
+            inv.setItem(i, item);
+        }
+
+        // If inventory is only one row, no need for anything else
+        if (size > 9) {
+            // Fill bottom
+            for (int i = size - 9; i < size; i++) {
+                inv.setItem(i, item);
+            }
+
+            // Fill sides
+            for (int i = 2; i <= rows - 1; i++) {
+                int[] slots = new int[]{i * 9 - 1, (i - 1) * 9};
+                inv.setItem(slots[0], item);
+                inv.setItem(slots[1], item);
```

Comment:
Why on earth are you allocating an array here??
```suggestion
                inv.setItem(i * 9 - 1, item);
                inv.setItem((i - 1) * 9, item);
```